blazor/file-manager/flat-data.md

```diff
@@ -0,0 +1,197 @@
+---
+layout: post
+title: Flat data rendering in Blazor FileManager component | Syncfusion
+description: Checkout and learn here all about flat data rendering in Syncfusion Blazor FileManager component and more.
+platform: Blazor
+control: File Manager
+documentation: ug
+---
+
+# Flat data rendering in Blazor FileManager component
+
+This section briefly explains how to render the FileManager component using a flat data object. With the latest enhancement, the File Manager component now employs a flat data object for rendering, eliminating the need to define `FileManagerAjaxSettings`. Instead, in the documentation code, a physical service is injected for file actions, and the corresponding file action's response can be retrieved from the locally injected service with the help of new event support for actions performed with the FileManager component.
+
+**Event information**
+
+Event Name | Description
+ ---  | ---
+OnRead | Gets or sets an event callback that will be invoked when the initial data is set and when the sub folders data are read.
+ItemsDeleting | Gets or sets an event callback that will be invoked before the delete operation takes place.
+FolderCreating | Gets or sets an event callback that will be invoked before the folder is being created.
+Searching | Gets or sets an event callback that will be invoked when the character is entered in input box for searching files or folders.
+ItemRenaming | Gets or sets an event callback that will be invoked when the file or folder is being renamed.
+ItemsMoving | Gets or sets an event callback that will be invoked when the file or folder is being cut or copied for the reason to be pasted in another path.
+ItemsUploaded | Gets or sets an event callback that will be invoked when the file or folder is uploaded.
+BeforeImageLoad | Gets or sets an event callback that will be invoked before sending the image request to the server.
+BeforeDownload  | Gets or sets an event callback that will be invoked before sending the download request to the server.
```

Comment:
Deleted, Moved and Uploading events are missing, also hyperlink the API reference for event names.

Reply:
@keerthanaRajendran as per the suggestion included a end event details and provided a link for new events. https://github.com/syncfusion-content/blazor-docs/pull/4076/commits/aba37ec5c56870af59b35a357c7d63f552140496